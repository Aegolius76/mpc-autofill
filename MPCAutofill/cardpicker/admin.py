from django.contrib import admin

from .models import Card, Cardback, DFCPair, Source, Token


# Register your models here.
@admin.register(Card)
class AdminCard(admin.ModelAdmin):
    list_display = ("name", "source", "dpi", "date")


@admin.register(Cardback)
class AdminCardback(admin.ModelAdmin):
    list_display = ("name", "source", "dpi", "date")


@admin.register(Token)
class AdminToken(admin.ModelAdmin):
    list_display = ("name", "source", "dpi", "date")


@admin.register(DFCPair)
class AdminDFCPair(admin.ModelAdmin):
    list_display = ("front", "back")


@admin.register(Source)
class AdminSource(admin.ModelAdmin):
    list_display = ("id", "drive_id", "contribution", "description")

    def contribution(self, obj):
        qty_all, qty_cards, qty_cardbacks, qty_tokens, avgdpi = obj.count()

        return "{} images - {} cards, {} cardbacks, and {} tokens @ {} DPI on average".format(
            qty_all, qty_cards, qty_cardbacks, qty_tokens, avgdpi
        )
