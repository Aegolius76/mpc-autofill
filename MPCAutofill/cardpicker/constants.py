DFC_URL = "https://api.scryfall.com/cards/search?q=is:dfc -layout:art_series -layout:double_faced_token -is:reversible"
MELD_URL = "https://api.scryfall.com/cards/search?q=is:meld"

DATE_FORMAT = "jS F, Y"

PAGE_SIZE = 10
