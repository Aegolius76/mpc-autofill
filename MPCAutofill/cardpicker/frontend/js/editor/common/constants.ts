export type CardType = "CARD" | "CARDBACK" | "TOKEN";
export const Card: CardType = "CARD";
export const Cardback: CardType = "CARDBACK";
export const Token: CardType = "TOKEN";

export const CardTypePrefixes = {
  "": Card,
  b: Cardback,
  t: Token,
};

export type Faces = "front" | "back";
export const Front: Faces = "front";
export const Back: Faces = "back";

export interface SearchQuery {
  query: string | null;
  card_type: CardType;
}

export const MinimumDPI = 0;
export const MaximumDPI = 1500;
export const DPIStep = 50;
