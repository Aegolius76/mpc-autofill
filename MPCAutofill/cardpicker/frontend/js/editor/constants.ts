export enum CardTypes {
  Card = "CARD",
  Cardback = "CARDBACK",
  Token = "TOKEN",
}

export interface SearchQuery {
  query: string;
  card_type: CardTypes;
}

export type Faces = "front" | "back";
export const Front: Faces = "front";
export const Back: Faces = "back";

export const CardTypePrefixes = {
  "": CardTypes.Card,
  b: CardTypes.Cardback,
  t: CardTypes.Token,
};
