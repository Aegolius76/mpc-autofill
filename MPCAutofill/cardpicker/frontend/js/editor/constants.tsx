export enum CardTypes {
  Card = "CARD",
  Cardback = "CARDBACK",
  Token = "TOKEN",
}

export interface SearchQuery {
  query: string;
  card_type: CardTypes;
}

export enum Faces {
  Front = "front",
  Back = "back",
}
