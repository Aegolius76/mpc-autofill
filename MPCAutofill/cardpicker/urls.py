from django.urls import path, re_path

from . import views

urlpatterns = [
    path("", views.index, name="index"),
    path("contributions", views.contributions, name="contributions"),
    path("patrons", views.patrons, name="patrons"),
    path("new", views.new_cards, name="new"),
    path("legal", views.legal, name="legal"),
    re_path(r"^ajax/getnew/$", views.search_new_page, name="getnew"),
    path("guide", views.guide, name="guide"),
    path("ajax/status/", views.elasticsearch_status, name="status"),
    # region old API
    re_path(r"^ajax/search/$", views.search_individual, name="search"),
    re_path(r"^ajax/msearch/$", views.search_multiple, name="msearch"),
    re_path(r"^ajax/text/$", views.insert_text, name="insert_text"),
    re_path(r"^ajax/xml/$", views.insert_xml, name="xml"),
    re_path(r"^ajax/link/$", views.insert_link, name="insert_link"),
    path("review", views.review, name="review"),
    path("input_xml", views.input_xml, name="input_xml"),
    path("input_csv", views.input_csv, name="input_csv"),
    path("input_link", views.input_link, name="input_link"),
    # endregion
    # region new API
    path("editor", views.editor, name="editor"),
    path("2/searchResults/", views.api_function_1, name="api_function_1"),
    path("2/cards/", views.api_function_2, name="api_function_2"),
    path("2/sources/", views.api_function_3, name="api_function_3"),
    path("2/DFCPairs/", views.api_function_4, name="api_function_4"),
    path("2/cardstocks/", views.api_function_5, name="api_function_5"),
    path("2/cardbacks/", views.api_function_6, name="api_function_6"),
    path("2/importSites/", views.api_function_7, name="api_function_7"),
    path("2/importSiteDecklist/", views.api_function_8, name="api_function_8"),  # TODO: rename this
    path("2/placeholderText/", views.api_function_9, name="api_function_9"),
    path("2/contributions/", views.api_function_10, name="api_function_10"),
    path("2/info/", views.api_function_11, name="api_function_11"),
    path("2/searchEngineHealth/", views.api_function_12, name="api_function_12"),
    # endregion
]
