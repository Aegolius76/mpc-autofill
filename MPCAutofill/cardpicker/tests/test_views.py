import pytest
from requests import Response
from syrupy import SnapshotAssertion

from django.urls import reverse

from cardpicker import views
from cardpicker.tests.constants import Cards, Decks


def snapshot_response(response: Response, snapshot: SnapshotAssertion):
    try:
        assert {"status_code": response.status_code, "json": response.json()} == snapshot
    except ValueError:  # non-json response
        assert {"status_code": response.status_code, "content": response.content} == snapshot


class TestPostSearchResults:
    @pytest.fixture(autouse=True)
    def autouse_populated_database(self, populated_database):
        pass


class TestPostCards:
    @pytest.fixture(autouse=True)
    def autouse_populated_database(self, populated_database):
        pass

    def test_get_single_card(self, client, snapshot):
        response = client.post(
            reverse(views.post_cards),
            {"card_identifiers": [Cards.GOBLIN.value.identifier]},
            content_type="application/json",
        )
        snapshot_response(response, snapshot)

    def test_get_multiple_cards(self, client, snapshot):
        response = client.post(
            reverse(views.post_cards),
            {"card_identifiers": [Cards.GOBLIN.value.identifier, Cards.DELVER_OF_SECRETS.value.identifier]},
            content_type="application/json",
        )
        snapshot_response(response, snapshot)

    def test_request_card_not_in_the_database(self, client, snapshot):
        response = client.post(
            reverse(views.post_cards),
            {"card_identifiers": [Cards.GOBLIN.value.identifier, "i don't exist in the database"]},
            content_type="application/json",
        )
        snapshot_response(response, snapshot)

    def test_response_to_malformed_card_identifier_list(self, client, snapshot):
        # TODO: this should probably return a different response for a malformed json body
        response = client.post(
            reverse(views.post_cards),
            {"card_identifiers": "i should be a list but i ain't"},
            content_type="application/json",
        )
        snapshot_response(response, snapshot)
        assert response.status_code == 400

    def test_response_to_malformed_json_body(self, client, snapshot):
        # TODO: this should probably return a different response for a malformed json body
        response = client.post(
            reverse(views.post_cards), {"test": "i should be a json body but i ain't"}, content_type="application/json"
        )
        snapshot_response(response, snapshot)
        assert response.status_code == 400


class TestGetSources:
    def test_get_multiple_sources(self, client, snapshot, all_sources):
        response = client.get(reverse(views.get_sources))
        snapshot_response(response, snapshot)


class TestGetDFCPairs:
    def test_get_multiple_rows(self, client, snapshot, dfc_pairs):
        response = client.get(reverse(views.get_dfc_pairs))
        snapshot_response(response, snapshot)


class TestGetCardbacks:
    @pytest.fixture(autouse=True)
    def autouse_populated_database(self, populated_database):
        pass

    def test_get_multiple_rows(self, client, snapshot):
        response = client.get(reverse(views.get_cardbacks))
        snapshot_response(response, snapshot)


class TestGetImportSites:
    def test_get_multiple_sites(self, client, snapshot):
        response = client.get(reverse(views.get_import_sites))
        snapshot_response(response, snapshot)


class TestPostImportSiteDecklist:
    @pytest.fixture(autouse=True)
    def autouse_populated_database(self, populated_database):
        pass

    @pytest.mark.parametrize("url", [x.value for x in Decks])
    def test_valid_url(self, client, snapshot, url):
        response = client.post(reverse(views.post_import_site_decklist), {"url": url}, content_type="application/json")
        snapshot_response(response, snapshot)

    def test_invalid_url(self, client, snapshot):
        response = client.post(
            reverse(views.post_import_site_decklist), {"url": "https://garbage.com"}, content_type="application/json"
        )
        snapshot_response(response, snapshot)
        assert response.status_code == 400

    def test_malformed_json_body(self, client, snapshot):
        response = client.post(
            reverse(views.post_import_site_decklist),
            {"test": "garbage and garbage accessories"},
            content_type="application/json",
        )
        snapshot_response(response, snapshot)
        assert response.status_code == 400

    def test_get_request(self, client, snapshot):
        response = client.get(reverse(views.post_import_site_decklist))
        snapshot_response(response, snapshot)
        assert response.status_code == 400


class TestGetSampleCards:
    def test_get_five_sample_cards(
        self,
        client,
        django_settings,
        elasticsearch,
        all_sources,
        # the endpoint should consistently return exactly these five cards (in a random order)
        brainstorm,
        island,
        island_classical,
        past_in_flames_1,
        goblin,
        snapshot,
    ):
        response = client.get(reverse(views.get_sample_cards))
        assert response.status_code == 200
        json_body = response.json()
        assert set(json_body.keys()) == {"cards"}
        assert set(json_body["cards"].keys()) == {"CARD", "TOKEN"}
        # the view returns a list of cards, but the order of the cards is deliberately random
        # keying the data by card name in this way should result in deterministic snapshotting
        assert {
            card_type: {row["name"]: row for row in sorted(cards, key=lambda x: x["name"])}
            for card_type, cards in json_body["cards"].items()
        } == snapshot

    def test_get_no_cards(self, client, django_settings, elasticsearch, all_sources, snapshot):
        response = client.get(reverse(views.get_sample_cards))
        assert response.status_code == 200
        assert response.json()["cards"] == {"CARD": [], "TOKEN": []}

    def test_get_three_cards_one_token(
        self,
        client,
        django_settings,
        elasticsearch,
        all_sources,
        brainstorm,
        island,
        island_classical,
        goblin,
        snapshot,
    ):
        response = client.get(reverse(views.get_sample_cards))
        assert response.status_code == 200
        json_body = response.json()
        assert len(json_body["cards"]["CARD"]) == 3
        assert len(json_body["cards"]["TOKEN"]) == 1

    def test_get_four_cards_zero_tokens(
        self,
        client,
        django_settings,
        elasticsearch,
        all_sources,
        brainstorm,
        island,
        island_classical,
        past_in_flames_1,
        snapshot,
    ):
        response = client.get(reverse(views.get_sample_cards))
        assert response.status_code == 200
        json_body = response.json()
        assert len(json_body["cards"]["CARD"]) == 4
        assert len(json_body["cards"]["TOKEN"]) == 0


class TestGetContributions:
    def test_get_multiple_rows(self, client, snapshot, all_sources, all_cards):
        response = client.get(reverse(views.get_contributions))
        snapshot_response(response, snapshot)

    def test_get_one_row(self, client, snapshot, example_drive_1, island, island_classical):
        response = client.get(reverse(views.get_contributions))
        snapshot_response(response, snapshot)

    def test_get_source_with_no_cards(self, client, snapshot, all_sources):
        response = client.get(reverse(views.get_contributions))
        snapshot_response(response, snapshot)

    def test_with_no_sources(self, client, snapshot, db):
        response = client.get(reverse(views.get_contributions))
        snapshot_response(response, snapshot)


class TestGetInfo:
    @pytest.fixture(autouse=True)
    def autouse_populated_database(self, populated_database):
        pass

    # TODO: write tests


class TestGetSearchEngineHealth:
    def test_elasticsearch_healthy(self, client, django_settings, elasticsearch, snapshot):
        response = client.get(reverse(views.get_search_engine_health))
        snapshot_response(response, snapshot)
        assert response.json()["online"] is True

    # TODO: consider how to test elasticsearch being unhealthy
