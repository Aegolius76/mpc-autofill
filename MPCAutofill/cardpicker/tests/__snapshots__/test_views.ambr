# name: TestGetCardbacks.test_get_multiple_rows
  dict({
    'json': dict({
      'cardbacks': list([
        '1JtXL6Ca9nQkvhwZZRR9ZuKA9_DzsFf1V',
        '1oigI6wz0zA--pNMuExKTs40kBNH6VRP_',
      ]),
    }),
    'status_code': 200,
  })
# ---
# name: TestGetContributions.test_get_multiple_rows
  dict({
    'json': dict({
      'card_count_by_type': dict({
        'CARD': 10,
        'CARDBACK': 2,
        'TOKEN': 1,
      }),
      'sources': list([
        dict({
          'avgdpi': '800.00',
          'description': 'Description for example_drive_1',
          'external_link': 'https://drive.google.com/open?id=1Fu2nEymZhCpOOZkfF0XoZsVqdIWmPdNq',
          'identifier': '1Fu2nEymZhCpOOZkfF0XoZsVqdIWmPdNq',
          'name': 'Example Drive 1',
          'qty_cardbacks': '2',
          'qty_cards': '9',
          'qty_tokens': '1',
          'size': '0.00 GB',
          'source_type': 'Google Drive',
        }),
        dict({
          'avgdpi': '800.00',
          'description': 'Description for example_drive_2',
          'external_link': None,
          'identifier': '18wZl7T9DU_lf2X5xYFiyH6pATVy8ZlOd',
          'name': 'Example Drive 2',
          'qty_cardbacks': '0',
          'qty_cards': '1',
          'qty_tokens': '0',
          'size': '0.00 GB',
          'source_type': 'Google Drive',
        }),
      ]),
      'total_database_size': 1300,
    }),
    'status_code': 200,
  })
# ---
# name: TestGetContributions.test_get_one_row
  dict({
    'json': dict({
      'card_count_by_type': dict({
        'CARD': 2,
        'CARDBACK': 0,
        'TOKEN': 0,
      }),
      'sources': list([
        dict({
          'avgdpi': '800.00',
          'description': 'Description for example_drive_1',
          'external_link': 'https://drive.google.com/open?id=1Fu2nEymZhCpOOZkfF0XoZsVqdIWmPdNq',
          'identifier': '1Fu2nEymZhCpOOZkfF0XoZsVqdIWmPdNq',
          'name': 'Example Drive 1',
          'qty_cardbacks': '0',
          'qty_cards': '2',
          'qty_tokens': '0',
          'size': '0.00 GB',
          'source_type': 'Google Drive',
        }),
      ]),
      'total_database_size': 200,
    }),
    'status_code': 200,
  })
# ---
# name: TestGetContributions.test_get_source_with_no_cards
  dict({
    'json': dict({
      'card_count_by_type': dict({
        'CARD': 0,
        'CARDBACK': 0,
        'TOKEN': 0,
      }),
      'sources': list([
        dict({
          'avgdpi': 0,
          'description': 'Description for example_drive_1',
          'external_link': 'https://drive.google.com/open?id=1Fu2nEymZhCpOOZkfF0XoZsVqdIWmPdNq',
          'identifier': '1Fu2nEymZhCpOOZkfF0XoZsVqdIWmPdNq',
          'name': 'Example Drive 1',
          'qty_cardbacks': '0',
          'qty_cards': '0',
          'qty_tokens': '0',
          'size': '0.00 GB',
          'source_type': 'Google Drive',
        }),
        dict({
          'avgdpi': 0,
          'description': 'Description for example_drive_2',
          'external_link': None,
          'identifier': '18wZl7T9DU_lf2X5xYFiyH6pATVy8ZlOd',
          'name': 'Example Drive 2',
          'qty_cardbacks': '0',
          'qty_cards': '0',
          'qty_tokens': '0',
          'size': '0.00 GB',
          'source_type': 'Google Drive',
        }),
      ]),
      'total_database_size': 0,
    }),
    'status_code': 200,
  })
# ---
# name: TestGetContributions.test_with_no_sources
  dict({
    'json': dict({
      'card_count_by_type': dict({
        'CARD': 0,
        'CARDBACK': 0,
        'TOKEN': 0,
      }),
      'sources': list([
      ]),
      'total_database_size': 0,
    }),
    'status_code': 200,
  })
# ---
# name: TestGetDFCPairs.test_get_multiple_rows
  dict({
    'json': dict({
      'dfc_pairs': dict({
        'Delver of Secrets': 'Insectile Aberration',
        'Huntmaster of the Fells': 'Ravager of the Fells',
      }),
    }),
    'status_code': 200,
  })
# ---
# name: TestGetImportSites.test_get_multiple_sites
  dict({
    'json': dict({
      'import_sites': list([
        dict({
          'name': 'Aetherhub',
          'url': 'https://aetherhub.com',
        }),
        dict({
          'name': 'Archidekt',
          'url': 'https://archidekt.com',
        }),
        dict({
          'name': 'CubeCobra',
          'url': 'https://cubecobra.com',
        }),
        dict({
          'name': 'Deckstats',
          'url': 'https://deckstats.net',
        }),
        dict({
          'name': 'MagicVille',
          'url': 'https://magic-ville.com',
        }),
        dict({
          'name': 'ManaStack',
          'url': 'https://manastack.com',
        }),
        dict({
          'name': 'Moxfield',
          'url': 'https://www.moxfield.com',
        }),
        dict({
          'name': 'MTGGoldfish',
          'url': 'https://www.mtggoldfish.com',
        }),
        dict({
          'name': 'Scryfall',
          'url': 'https://scryfall.com',
        }),
        dict({
          'name': 'TappedOut',
          'url': 'https://tappedout.net',
        }),
        dict({
          'name': 'TCGPlayer',
          'url': 'https://decks.tcgplayer.com/',
        }),
      ]),
    }),
    'status_code': 200,
  })
# ---
# name: TestGetSampleCards.test_get_five_sample_cards
  dict({
    'CARD': dict({
      'Brainstorm': dict({
        'card_type': 'CARD',
        'date': '1st January, 2023',
        'download_link': 'https://drive.google.com/uc?id=1c4M-sK9gd0Xju0NXCPtqeTW_DQTldVU5&export=download',
        'dpi': 800,
        'extension': 'png',
        'identifier': '1c4M-sK9gd0Xju0NXCPtqeTW_DQTldVU5',
        'medium_thumbnail_url': 'https://drive.google.com/thumbnail?sz=w800-h800&id=1c4M-sK9gd0Xju0NXCPtqeTW_DQTldVU5',
        'name': 'Brainstorm',
        'priority': 2,
        'searchq': 'brainstorm',
        'size': 100,
        'small_thumbnail_url': 'https://drive.google.com/thumbnail?sz=w400-h400&id=1c4M-sK9gd0Xju0NXCPtqeTW_DQTldVU5',
        'source': 'example_drive_1',
        'source_id': 0,
        'source_name': 'Example Drive 1',
        'source_type': 'Google Drive',
        'source_verbose': 'Example Drive 1 but verbose',
      }),
      'Island': dict({
        'card_type': 'CARD',
        'date': '1st January, 2023',
        'download_link': 'https://drive.google.com/uc?id=1IDtqSjJ4Yo45AnNA4SplOiN7ewibifMa&export=download',
        'dpi': 800,
        'extension': 'png',
        'identifier': '1IDtqSjJ4Yo45AnNA4SplOiN7ewibifMa',
        'medium_thumbnail_url': 'https://drive.google.com/thumbnail?sz=w800-h800&id=1IDtqSjJ4Yo45AnNA4SplOiN7ewibifMa',
        'name': 'Island',
        'priority': 7,
        'searchq': 'island',
        'size': 100,
        'small_thumbnail_url': 'https://drive.google.com/thumbnail?sz=w400-h400&id=1IDtqSjJ4Yo45AnNA4SplOiN7ewibifMa',
        'source': 'example_drive_1',
        'source_id': 0,
        'source_name': 'Example Drive 1',
        'source_type': 'Google Drive',
        'source_verbose': 'Example Drive 1 but verbose',
      }),
      'Island (William Bradford)': dict({
        'card_type': 'CARD',
        'date': '1st January, 2023',
        'download_link': 'https://drive.google.com/uc?id=1HsvTYs1jFGe1c8U1PnNZ9aB8jkAW7KU0&export=download',
        'dpi': 800,
        'extension': 'png',
        'identifier': '1HsvTYs1jFGe1c8U1PnNZ9aB8jkAW7KU0',
        'medium_thumbnail_url': 'https://drive.google.com/thumbnail?sz=w800-h800&id=1HsvTYs1jFGe1c8U1PnNZ9aB8jkAW7KU0',
        'name': 'Island (William Bradford)',
        'priority': 6,
        'searchq': 'island',
        'size': 100,
        'small_thumbnail_url': 'https://drive.google.com/thumbnail?sz=w400-h400&id=1HsvTYs1jFGe1c8U1PnNZ9aB8jkAW7KU0',
        'source': 'example_drive_1',
        'source_id': 0,
        'source_name': 'Example Drive 1',
        'source_type': 'Google Drive',
        'source_verbose': 'Example Drive 1 but verbose',
      }),
      'Past in Flames': dict({
        'card_type': 'CARD',
        'date': '1st January, 2023',
        'download_link': 'https://drive.google.com/uc?id=1UPdh7J7hScg4ZnxSPJ-EeBYHLp2s3Oz1&export=download',
        'dpi': 800,
        'extension': 'png',
        'identifier': '1UPdh7J7hScg4ZnxSPJ-EeBYHLp2s3Oz1',
        'medium_thumbnail_url': 'https://drive.google.com/thumbnail?sz=w800-h800&id=1UPdh7J7hScg4ZnxSPJ-EeBYHLp2s3Oz1',
        'name': 'Past in Flames',
        'priority': 2,
        'searchq': 'past in flames',
        'size': 100,
        'small_thumbnail_url': 'https://drive.google.com/thumbnail?sz=w400-h400&id=1UPdh7J7hScg4ZnxSPJ-EeBYHLp2s3Oz1',
        'source': 'example_drive_1',
        'source_id': 0,
        'source_name': 'Example Drive 1',
        'source_type': 'Google Drive',
        'source_verbose': 'Example Drive 1 but verbose',
      }),
    }),
    'TOKEN': dict({
      'Goblin': dict({
        'card_type': 'TOKEN',
        'date': '1st January, 2023',
        'download_link': 'https://drive.google.com/uc?id=1V5E0avDmNyEUuFfYwx3nA05aj-1HY0rA&export=download',
        'dpi': 800,
        'extension': 'png',
        'identifier': '1V5E0avDmNyEUuFfYwx3nA05aj-1HY0rA',
        'medium_thumbnail_url': 'https://drive.google.com/thumbnail?sz=w800-h800&id=1V5E0avDmNyEUuFfYwx3nA05aj-1HY0rA',
        'name': 'Goblin',
        'priority': 2,
        'searchq': 'goblin',
        'size': 100,
        'small_thumbnail_url': 'https://drive.google.com/thumbnail?sz=w400-h400&id=1V5E0avDmNyEUuFfYwx3nA05aj-1HY0rA',
        'source': 'example_drive_1',
        'source_id': 0,
        'source_name': 'Example Drive 1',
        'source_type': 'Google Drive',
        'source_verbose': 'Example Drive 1 but verbose',
      }),
    }),
  })
# ---
# name: TestGetSearchEngineHealth.test_elasticsearch_healthy
  dict({
    'json': dict({
      'online': True,
    }),
    'status_code': 200,
  })
# ---
# name: TestGetSources.test_get_multiple_sources
  dict({
    'json': dict({
      'results': dict({
        '0': dict({
          'description': 'Description for example_drive_1',
          'external_link': 'https://drive.google.com/open?id=1Fu2nEymZhCpOOZkfF0XoZsVqdIWmPdNq',
          'identifier': '1Fu2nEymZhCpOOZkfF0XoZsVqdIWmPdNq',
          'key': 'example_drive_1',
          'name': 'Example Drive 1',
          'pk': 0,
          'source_type': 'Google Drive',
        }),
        '1': dict({
          'description': 'Description for example_drive_2',
          'external_link': None,
          'identifier': '18wZl7T9DU_lf2X5xYFiyH6pATVy8ZlOd',
          'key': 'example_drive_2',
          'name': 'Example Drive 2',
          'pk': 1,
          'source_type': 'Google Drive',
        }),
      }),
    }),
    'status_code': 200,
  })
# ---
# name: TestPostCards.test_get_multiple_cards
  dict({
    'json': dict({
      'results': dict({
        '17fopRCNRge72U8Hac8pApHZtEalx5kHy': dict({
          'card_type': 'CARD',
          'date': '1st January, 2023',
          'download_link': 'https://drive.google.com/uc?id=17fopRCNRge72U8Hac8pApHZtEalx5kHy&export=download',
          'dpi': 800,
          'extension': 'png',
          'identifier': '17fopRCNRge72U8Hac8pApHZtEalx5kHy',
          'medium_thumbnail_url': 'https://drive.google.com/thumbnail?sz=w800-h800&id=17fopRCNRge72U8Hac8pApHZtEalx5kHy',
          'name': 'Delver of Secrets',
          'priority': 2,
          'searchq': 'delver of secrets',
          'size': 100,
          'small_thumbnail_url': 'https://drive.google.com/thumbnail?sz=w400-h400&id=17fopRCNRge72U8Hac8pApHZtEalx5kHy',
          'source': 'example_drive_1',
          'source_id': 0,
          'source_name': 'Example Drive 1',
          'source_type': 'Google Drive',
          'source_verbose': 'Example Drive 1 but verbose',
        }),
        '1V5E0avDmNyEUuFfYwx3nA05aj-1HY0rA': dict({
          'card_type': 'TOKEN',
          'date': '1st January, 2023',
          'download_link': 'https://drive.google.com/uc?id=1V5E0avDmNyEUuFfYwx3nA05aj-1HY0rA&export=download',
          'dpi': 800,
          'extension': 'png',
          'identifier': '1V5E0avDmNyEUuFfYwx3nA05aj-1HY0rA',
          'medium_thumbnail_url': 'https://drive.google.com/thumbnail?sz=w800-h800&id=1V5E0avDmNyEUuFfYwx3nA05aj-1HY0rA',
          'name': 'Goblin',
          'priority': 2,
          'searchq': 'goblin',
          'size': 100,
          'small_thumbnail_url': 'https://drive.google.com/thumbnail?sz=w400-h400&id=1V5E0avDmNyEUuFfYwx3nA05aj-1HY0rA',
          'source': 'example_drive_1',
          'source_id': 0,
          'source_name': 'Example Drive 1',
          'source_type': 'Google Drive',
          'source_verbose': 'Example Drive 1 but verbose',
        }),
      }),
    }),
    'status_code': 200,
  })
# ---
# name: TestPostCards.test_get_single_card
  dict({
    'json': dict({
      'results': dict({
        '1V5E0avDmNyEUuFfYwx3nA05aj-1HY0rA': dict({
          'card_type': 'TOKEN',
          'date': '1st January, 2023',
          'download_link': 'https://drive.google.com/uc?id=1V5E0avDmNyEUuFfYwx3nA05aj-1HY0rA&export=download',
          'dpi': 800,
          'extension': 'png',
          'identifier': '1V5E0avDmNyEUuFfYwx3nA05aj-1HY0rA',
          'medium_thumbnail_url': 'https://drive.google.com/thumbnail?sz=w800-h800&id=1V5E0avDmNyEUuFfYwx3nA05aj-1HY0rA',
          'name': 'Goblin',
          'priority': 2,
          'searchq': 'goblin',
          'size': 100,
          'small_thumbnail_url': 'https://drive.google.com/thumbnail?sz=w400-h400&id=1V5E0avDmNyEUuFfYwx3nA05aj-1HY0rA',
          'source': 'example_drive_1',
          'source_id': 0,
          'source_name': 'Example Drive 1',
          'source_type': 'Google Drive',
          'source_verbose': 'Example Drive 1 but verbose',
        }),
      }),
    }),
    'status_code': 200,
  })
# ---
# name: TestPostCards.test_request_card_not_in_the_database
  dict({
    'json': dict({
      'results': dict({
        '1V5E0avDmNyEUuFfYwx3nA05aj-1HY0rA': dict({
          'card_type': 'TOKEN',
          'date': '1st January, 2023',
          'download_link': 'https://drive.google.com/uc?id=1V5E0avDmNyEUuFfYwx3nA05aj-1HY0rA&export=download',
          'dpi': 800,
          'extension': 'png',
          'identifier': '1V5E0avDmNyEUuFfYwx3nA05aj-1HY0rA',
          'medium_thumbnail_url': 'https://drive.google.com/thumbnail?sz=w800-h800&id=1V5E0avDmNyEUuFfYwx3nA05aj-1HY0rA',
          'name': 'Goblin',
          'priority': 2,
          'searchq': 'goblin',
          'size': 100,
          'small_thumbnail_url': 'https://drive.google.com/thumbnail?sz=w400-h400&id=1V5E0avDmNyEUuFfYwx3nA05aj-1HY0rA',
          'source': 'example_drive_1',
          'source_id': 0,
          'source_name': 'Example Drive 1',
          'source_type': 'Google Drive',
          'source_verbose': 'Example Drive 1 but verbose',
        }),
      }),
    }),
    'status_code': 200,
  })
# ---
# name: TestPostCards.test_response_to_malformed_card_identifier_list
  dict({
    'json': dict({
      'results': dict({
      }),
    }),
    'status_code': 200,
  })
# ---
# name: TestPostCards.test_response_to_malformed_json_body
  dict({
    'json': dict({
      'results': dict({
      }),
    }),
    'status_code': 200,
  })
# ---
# name: TestPostImportSiteDecklist.test_get_request
  dict({
    'content': b'Expected POST request.',
    'status_code': 400,
  })
# ---
# name: TestPostImportSiteDecklist.test_invalid_url
  dict({
    'content': b'The specified decklist URL does not match any known import sites.',
    'status_code': 400,
  })
# ---
# name: TestPostImportSiteDecklist.test_malformed_json_body
  dict({
    'content': b'No decklist URL provided.',
    'status_code': 400,
  })
# ---
# name: TestPostImportSiteDecklist.test_valid_url[https://aetherhub.com/Deck/test-796905]
  dict({
    'json': dict({
      'cards': '''
        4 Brainstorm
        3 Past in Flames
        1 Delver of Secrets
  
      ''',
    }),
    'status_code': 200,
  })
# ---
# name: TestPostImportSiteDecklist.test_valid_url[https://archidekt.com/decks/3380653]
  dict({
    'json': dict({
      'cards': '''
        1 Delver of Secrets // Insectile Aberration
        4 Brainstorm
        3 Past in Flames
  
      ''',
    }),
    'status_code': 200,
  })
# ---
# name: TestPostImportSiteDecklist.test_valid_url[https://cubecobra.com/cube/overview/2fj4]
  dict({
    'json': dict({
      'cards': '''
        Delver of Secrets
        Brainstorm
        Brainstorm
        Brainstorm
        Brainstorm
        Past in Flames
        Past in Flames
        Past in Flames
        
        
  
      ''',
    }),
    'status_code': 200,
  })
# ---
# name: TestPostImportSiteDecklist.test_valid_url[https://decks.tcgplayer.com/magic/standard/mpc-autofill/test/1398367]
  dict({
    'json': dict({
      'cards': '''
        1 Delver of Secrets
        4 Brainstorm
        3 Past in Flames
  
      ''',
    }),
    'status_code': 200,
  })
# ---
# name: TestPostImportSiteDecklist.test_valid_url[https://deckstats.net/decks/216625/2754468-test]
  dict({
    'json': dict({
      'cards': '''
        4 Brainstorm
        1 Delver of Secrets // Insectile Aberration
        3 Past in Flames
  
      ''',
    }),
    'status_code': 200,
  })
# ---
# name: TestPostImportSiteDecklist.test_valid_url[https://magic-ville.com/fr/decks/showdeck?ref=948045]
  dict({
    'json': dict({
      'cards': '''
        3 Past in Flames
        4 Brainstorm
        1 Delver of Secrets
  
      ''',
    }),
    'status_code': 200,
  })
# ---
# name: TestPostImportSiteDecklist.test_valid_url[https://manastack.com/deck/test-426]
  dict({
    'json': dict({
      'cards': '''
        4 Brainstorm
        3 Past in Flames
        1 Delver of Secrets
  
      ''',
    }),
    'status_code': 200,
  })
# ---
# name: TestPostImportSiteDecklist.test_valid_url[https://scryfall.com/@mpcautofill/decks/71bb2d40-c922-4a01-a63e-7ba2dde29a5c]
  dict({
    'json': dict({
      'cards': '''
        4 Brainstorm
        1 Delver of Secrets
        3 Past in Flames
        
  
      ''',
    }),
    'status_code': 200,
  })
# ---
# name: TestPostImportSiteDecklist.test_valid_url[https://tappedout.net/mtg-decks/09-10-22-DoY-test]
  dict({
    'json': dict({
      'cards': '''
        4 Brainstorm
        1 Delver of Secrets
        3 Past in Flames
        
        
  
      ''',
    }),
    'status_code': 200,
  })
# ---
# name: TestPostImportSiteDecklist.test_valid_url[https://www.moxfield.com/decks/D42-or9pCk-uMi4XzRDziQ]
  dict({
    'json': dict({
      'cards': '''
        1 Delver of Secrets // Insectile Aberration
        4 Brainstorm
        3 Past in Flames
  
      ''',
    }),
    'status_code': 200,
  })
# ---
# name: TestPostImportSiteDecklist.test_valid_url[https://www.mtggoldfish.com/deck/5149750]
  dict({
    'json': dict({
      'cards': '''
        4 Brainstorm
        1 Delver of Secrets
        3 Past in Flames
        
  
      ''',
    }),
    'status_code': 200,
  })
# ---
